### PR TITLE
Profiling ERC20 contract

### PR DESCRIPTION
**Pull request by baptiste-b-pegasys**
fixes https://github.com/ConsenSys/quorum-board/issues/84

Profiling by transfering of 1 wei token per transaction.

```
pragma solidity ^0.8.0;

import "@openzeppelin/contracts/token/ERC20/ERC20.sol";

contract MyCollectible is ERC20 {
    constructor() ERC20("MyCollectible", "MCO") {
    }
}
```

### CODE REVIEW

**Comment by baptiste-b-pegasys** [comment]
I added some graph for AWS stress test :
image
image

File: stresstest-aws/scripts/test/quorum-test-dashboard.json [hunk 1]
```diff
@@ -1462,51 +1473,1939 @@
...
+              "type": "tag"
+            },
+            {
+              "params": [
+                "null"
+              ],
+              "type": "fill"
+            }
+          ],
+          "measurement": "geth.eth/db/chaindata/ancient/size.gauge",
```
[the remark this answers is not in the document]
Comment (baptiste-b-pegasys) [reply stresstest-aws/scripts/test/quorum-test-dashboard.json:2510]: It is a metric from freezerTable, "db/chaindata" is the namespace.
FreezerTable is a table that will do some cold storage.
"ancient" is related to freeze/cold storage, I think.
Looking at the source for the namespace, chaindata is opened in eth.backend.go as chaindb and eth.chainDb seems not to be used a lot. I saw it is passed to the genesis for having the chainConfig.

File: jmeter-test/4node/deploy-fungible-expand-contract-private.jmx [hunk 2]
```diff
@@ -0,0 +1,1830 @@
...
+            <stringProp name="parameters"></stringProp>
+            <stringProp name="filename"></stringProp>
+            <stringProp name="script">import org.apache.jmeter.util.JMeterUtils;
+import org.apache.commons.lang3;
+
+int random = Integer.parseInt(vars.get(&quot;random&quot;));
+
+String randomInHex = Integer.toHexString(random);
+
+String data = &quot;0xa9059cbb000000000000000000000000&quot;;
```
[the remark this answers is not in the document]
Comment (baptiste-b-pegasys) [reply jmeter-test/4node/deploy-fungible-expand-contract-private.jmx:1194]: It's for calling the `transfer` function, which takes two arguments : an address to send the ERC20 token, and the amount.
If I transfert to the same address, the space of the map will not increase, so I randomly generate the addresses. (the balances map of the contract)

File: jmeter-test/4node/deploy-fungible-expand-contract-private.jmx [hunk 3]
```diff
@@ -0,0 +1,1830 @@
...
+String addr = &quot;ed9d02e382b34818e88b88a309c7fe71e65f419e&quot;;
+String addr2 = addr.substring(0, addr.length()-randomInHex.length()) + randomInHex;
+data += addr2.substring(0, addr.length()); // random address
+data += &quot;0000000000000000000000000000000000000000000000000000000000000001&quot;; // send 1 token
```
[the remark this answers is not in the document]
Comment (baptiste-b-pegasys) [reply jmeter-test/4node/deploy-fungible-expand-contract-private.jmx:1198]:
```suggestion
data += &quot;0000000000000000000000000000000000000000000000000000000000000001&quot;; // send minimal amount of ERC token (10^-18)
```

File: stresstest-aws/scripts/node/disk-usage.sh [hunk 4]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+while read -a LIST; \
+do timestamp=$(date +%s%N); \
+        printf '%s\n' 'disk-usage,folder="'${LIST[1]}'" bytes='${LIST[0]}' '$timestamp; \
```
Comment (dolanor) [inline stresstest-aws/scripts/node/disk-usage.sh:4]:
```suggestion
        printf 'disk-usage,folder="%s" bytes=%s %s\n', ${LIST[1]}, ${LIST[0]}, $timestamp; \
```
Wouldn't that make more sense if using printf?

**Review by ricardolyn — APPROVED** [review]

**Pull request merged**